src/ScreenResource.cc

```diff
@@ -76,8 +77,34 @@ setFromString(const char *strval) {
 
 } // end namespace FbTk
 
+template<>
+std::string FbTk::Resource<BScreen::WorkspaceNames>::getString() const {
+    std::string val;
+    for (size_t i = 0; i < m_value.size(); ++i) {
+        if (i > 0) val += ", ";
+	std::string chunk = m_value[i];
+	// Double all commas in workspace name so they can be restored
+	val += std::regex_replace(chunk, std::regex(","), ",,");
+    }
+    return val;
+}
 
-
+template<>
+void FbTk::Resource<BScreen::WorkspaceNames>::
+setFromString(const char* strval) {
+    m_value.clear();
+    std::string remainder = strval;
+    // Undoubled comma followed by a whitespace character separates names
```

Comment:
The resources use octals for utf-8, so maybe the literal comma should simply be encoded as \054 instead of the commawhitespaceorcommacommawhitespacegeneratescommawhitespace logic?

Reply:
will be moot for mgumz' redesign

Reply:
@gwhitney the mood on redesigning fluxbox things is slim.

the whole wayland-topic is also lingering around, the various config formats, is "advanced c++" really needed to achieve the slot/signal kind of behaviour from qt, is c++ really needed etc etc.

Reply:
ah, `moot` … interesting (i was first thinking of typo-of-mood). yes, exactly.

Reply:
Correct. This PR needs to be replaced by one that implements your proposed redesign of these parameters. However, there are still some questions in the last comment I made on this PR (https://github.com/fluxbox/fluxbox/pull/60#issuecomment-1510538083). When the warnings PR is merged and those questions are answered, I will be very happy to push a new PR to replace this one.